includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -431,15 +197,14 @@ public function process_payment( $order_id ) {
 				'invalid_phone_number'
 			);
 		}
-		$payment_intent_id         = isset( $_POST['wc_payment_intent_id'] ) ? wc_clean( wp_unslash( $_POST['wc_payment_intent_id'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.Missing
-		$amount                    = $order->get_total();
-		$currency                  = $order->get_currency();
-		$converted_amount          = WC_Payments_Utils::prepare_amount( $amount, $currency );
-		$payment_needed            = 0 < $converted_amount;
-		$selected_upe_payment_type = ! empty( $_POST['wcpay_selected_upe_payment_type'] ) ? wc_clean( wp_unslash( $_POST['wcpay_selected_upe_payment_type'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.Missing
-		$payment_type              = $this->is_payment_recurring( $order_id ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
-		$save_payment_method       = $payment_type->equals( Payment_Type::RECURRING() ) || ! empty( $_POST[ 'wc-' . $this->id . '-new-payment-method' ] ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
-		$payment_country           = ! empty( $_POST['wcpay_payment_country'] ) ? wc_clean( wp_unslash( $_POST['wcpay_payment_country'] ) ) : null; // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$payment_intent_id   = isset( $_POST['wc_payment_intent_id'] ) ? wc_clean( wp_unslash( $_POST['wc_payment_intent_id'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$amount              = $order->get_total();
+		$currency            = $order->get_currency();
+		$converted_amount    = WC_Payments_Utils::prepare_amount( $amount, $currency );
+		$payment_needed      = 0 < $converted_amount;
+		$payment_type        = $this->is_payment_recurring( $order_id ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
+		$save_payment_method = $payment_type->equals( Payment_Type::RECURRING() ) || ! empty( $_POST[ 'wc-' . $this->id . '-new-payment-method' ] ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		$payment_country     = ! empty( $_POST['wcpay_payment_country'] ) ? wc_clean( wp_unslash( $_POST['wcpay_payment_country'] ) ) : null; // phpcs:ignore WordPress.Security.NonceVerification.Missing
```

Comment:
This diff looks way bigger than it actually is: I just removed `$selected_upe_payment_type` variable and replaced it with `$this->stripe_id`, because with the latest implementation, we just [set the value of this param to be `$this->stripe_id`.](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/payment-methods/class-upe-split-payment-gateway.php#L295) No need to create this additional step/complexity, so we now use $this->stripe_id directly below (see line `253`).